### PR TITLE
Change reverselookup check

### PR DESCRIPTION
**Pull request by darthsiroftardis**
CHANGELOG:

-  During migration we purposefully set the reverselookup mode to be enabled, but it also possible that extent contracts would have the ownership mode set to minter and the minting mode set to installer only. This PR changes the logic to not error out in that case but result in a no-op allowing for migrating users to still get their receipts 
- Enforce consistency of underscores by replacing hyphens with underscores

### CODE REVIEW

File: tests/src/utility/constants.rs [hunk 1]
```diff
@@ -94,7 +94,7 @@ pub(crate) const CONTRACT_1_0_0_WASM: &str = "1_0_0/contract.wasm";
 pub(crate) const MINT_1_0_0_WASM: &str = "1_0_0/mint_call.wasm";
 pub(crate) const PAGE_SIZE: u64 = 1000;
 pub(crate) const UNMATCHED_HASH_COUNT: &str = "unmatched_hash_count";
-pub(crate) const PAGE_DICTIONARY_PREFIX: &str = "page-";
+pub(crate) const PAGE_DICTIONARY_PREFIX: &str = "page_";
```
Comment (gRoussac) [inline tests/src/utility/constants.rs:97]: Just a detail it's a matter of choice I guess but I would remove the separator from the prefix and replace occurrences of PAGE_DICTIONARY_PREFIX
```
 &format!("{PAGE_DICTIONARY_PREFIX}{SEP}{page_number}"),
 ```
 and 
 
 ```
 format!("{nft_receipt}{SEP}m{SEP}{PAGE_SIZE}{SEP}p{SEP}{page_table_entry}")
 ```
 one day you might want something like that
  ```
 "format!("{PAGE_DICTIONARY_PREFIX}{SEP}{page_table_entry}{SEP}{page_number}")
 ```
 
 When I run make lint I have another rule of formatting apparently.

File: contract/src/main.rs [hunk 2]
```diff
@@ -1386,56 +1386,54 @@ pub extern "C" fn updated_receipts() {
...
+                vec![false; page_table_width as usize],
+            );
+        }
+        let collection_name = utils::get_stored_value_with_user_errors::<String>(
+            COLLECTION_NAME,
+            NFTCoreError::MissingCollectionName,
+            NFTCoreError::InvalidCollectionName,
         );
+        let package_uref = storage::new_uref(utils::get_stored_value_with_user_errors::<String>(
+            &format!("cep78_{}", collection_name),
```
Comment (gRoussac) [inline contract/src/main.rs]: I would consider adding a const for the prefix too

```
pub(crate) const CEP_PREFIX: &str = "cep78";
...
&format!("{CEP_PREFIX}_{collection_name}")
```

File: contract/src/main.rs [hunk 3]
```diff
@@ -1386,56 +1386,54 @@ pub extern "C" fn updated_receipts() {
 
 #[no_mangle]
 pub extern "C" fn register_owner() {
-    if let OwnerReverseLookupMode::NoLookUp = utils::get_reporting_mode() {
```
Comment (gRoussac) [inline contract/src/main.rs:1389]: Before you were treating the exception NoLookUp and returning, now you surround with a big if with Complete, and it only affects burn.rs tests ? I would expect a change on that statement to trigger a test on InvalidReportingMode being returned but maybe it was not covered  or there is no point covering that.

**Review by gRoussac — APPROVED** [review]
I made some comments but they are not blocking your PR.

File: contract/src/main.rs (hunk 2 above)
Comment (darthsiroftardis) [reply contract/src/main.rs]: Added CEP-78 prefix

File: tests/src/utility/constants.rs (hunk 1 above)
Comment (darthsiroftardis) [reply tests/src/utility/constants.rs:97]: Meh, personally I like having the seperators be in the macro invocation

Comment (gRoussac) [reply tests/src/utility/constants.rs:97]: I understand but then consider removing it from the prefix

```
 &format!("{PAGE_DICTIONARY_PREFIX}_{page_number}"),
```

like 

```
pub const HASH_KEY_NAME_PREFIX: &str = "cep78_contract_package_";
pub const ACCESS_KEY_NAME_PREFIX: &str = "cep78_contract_package_access_";
```

Those trailing separators are not super conventional.

**Pull request merged**